MCLOUD-6215: Add examples in devdocs how to downgrade service versions (#7657)
magento/devdocs: src/cloud/project/project-conf-files_services.md

## Service values

> You must provide the _type_ values: _name_ and _version_. If the service uses persistent storage, then you must provide a _disk_ value. Use the following format:

Reviewer: ```suggestion
You must provide the following service configuration values:

-  Service ID is a unique identifier for the service
-  Service type _name_  (_mysql, redis, elasticsearch, or rabbitmq_)
-  Service type _version_ must specify a supported version of the specified service type.

If the service uses persistent storage, then you must provide a _disk_ value.

Use the following format:
```

> You must provide the _type_ values: _service-name_ and _version_. If the service uses persistent storage, then you must provide a _disk_ value. Use the following format:

Reviewer 2: Why did we change this to `service-name` here and in the YAML example below but leave it as just `name` in the documentation (eg, the heading on line 54 and the following two paragraphs [56 and 58]).

I think `name` is fine, but either way we should be consistent.

Author: Yes, need to review all examples, but not we have 2 `<name>` in this schema https://devdocs.magento.com/cloud/project/project-conf-files_services.html#service-values what is wrong because service names are limited (redis, mysql, elasticsearch and rabbitmq), when service id can have any value with alphanumeric characters. 
Probably we may add this as
`<service-id>:
    type: <name>:<version>
    disk: <value-MB>`

Author: @meker12 can you please suggest better naming for fields to avoid `<name>` for two different fields

Reviewer: service-id -- works. Added suggestions to PR.

> You must provide the _type_ values: _service-name_ and _version_. If the service uses persistent storage, then you must provide a _disk_ value. Use the following format:

Reviewer: ```suggestion
You must provide the _type_ values: _name_ and _version_. If the service uses persistent storage, then you must provide a _disk_ value. Use the following format:
```

```yaml

> <name>:

Reviewer: ```suggestion
<service-id>:
```

>     type: <service-name>:<version>

Reviewer: ```suggestion
    type: <name>:<version>
```

>     type: <service-name>:<version>

Reviewer: ```suggestion
    type: <name>:<version>
```

    disk: <value-MB>
```

> ### `name`

Reviewer: ```suggestion
### `service-id`
```

> The `name` value identifies the service in the project. You can only use lower case alphanumeric characters: `a` to `z` and `0` to `9`, such as `redis`.

Reviewer: ```suggestion
The `service-id` value identifies the service in the project. You can only use lower case alphanumeric characters: `a` to `z` and `0` to `9`, such as `redis`.
```

> This _name_ value is used in the [`relationships`]({{ site.baseurl }}/cloud/project/project-conf-files_magento-app.html#relationships) property of the `.magento.app.yaml` configuration file:

Reviewer: ```suggestion
This _service-id_ value is used in the [`relationships`]({{ site.baseurl }}/cloud/project/project-conf-files_magento-app.html#relationships) property of the `.magento.app.yaml` configuration file:
```

    type: mysql:10.2
```

> Use [`Service versions`]{Service versions} table to see supported services and their versions

Reviewer: ```suggestion
Use [`Service versions`](#Service versions) table to see supported services and their versions
```

   echo $MAGENTO_CLOUD_RELATIONSHIPS | base64 -d | json_pp
   ```

>    or using Cloud ece-tools

Reviewer: ```suggestion
   or, use the following {{ site.data.var.ct }} CLI command to view relationships:
```

{:.bs-callout-warning}
You must submit a support ticket to change the service configuration on Pro Production and Staging environments.

{:.bs-callout-info}
If you change the Elasticsearch service version, check the Elasticsearch composer package for compatibility with the new version. See [Elasticsearch software compatibility]({{ site.baseurl}}/cloud/project/project-conf-files_services-elastic.html#elasticsearch-software-compatibility).

### Upgrade installed service version

You can upgrade the installed service version by updating the service configuration in the `services.yaml` file.

1. Change the [`type`](#type) value for the service in the `.magento/services.yaml` file:

   > Original service definition

   ```yaml
   mysql:
       type: mysql:10.2
       disk: 2048
   ```

   > Updated service definition

   ```yaml
   mysql:
       type: mysql:10.3
       disk: 2048
   ```

1. Add, commit, and push your code changes.

   ```bash
   git add -A
   ```

   ```bash
   git commit -m "Upgrade MySQL from MariaDB 10.2 to 10.3."
   ```

   ```bash
   git push origin <branch-name>
   ```

> ### Downgrade version

Reviewer 2: I feel like we should add a callout here reminding users that this is a destructive process and will delete their existing database data.

Author: We have this information in the description of the service id field 
```
Renaming a service in the services.yaml file permanently removes the following:
 - The existing service before creating a new service with the new name you specify.
 - All existing data for the service is removed. We strongly recommend you snapshot your environment before you change the name of an existing service.
```
Probably we need to duplicate it

Reviewer: I clarified the first bullet, and added a warning after step 1 of the "Rename" procedure. See most recent commits.

You cannot downgrade an installed service directly. You have two options:

> -  Rename an existing service with the new version, which removes the existing service and adds a new one.

Reviewer: ```suggestion
-  Rename an existing service with the new version, which removes the existing service and data, and adds a new one.
```

-  Create a new service and save the data from the existing service.

When you change the service version, you must update the service configuration in the `services.yaml` file, and update the relationships in the `.magento.app.yaml` file.

{:.procedure}
To downgrade a service version by renaming an existing service:

> 1. Rename the existing service in the `.magento/services.yaml` file and change the version.

Reviewer: ```suggestion
1. Rename the existing service in the `.magento/services.yaml` file and change the version.

   {:.bs-callout-warning}
   Renaming an existing service replaces it and deletes all data. If you need to retain the data, create a new service instead of renaming the existing one.
```

   ```yaml
   mysql:
       type: mysql:10.2
       disk: 2048
   ```

   > Updated service definition

   ```yaml
   mysql:
       type: mysql:10.3
       disk: 2048
   ```

1. Add, commit, and push your code changes.

   ```bash
   git add -A
   ```

   ```bash
   git commit -m "Upgrade MySQL from MariaDB 10.2 to 10.3."
   ```

   ```bash
   git push origin <branch-name>
   ```

### Downgrade version

You cannot downgrade an installed service directly. You have two options:

-  Rename an existing service with the new version, which removes the existing service and data, and adds a new one.

-  Create a new service and save the data from the existing service.

When you change the service version, you must update the service configuration in the `services.yaml` file, and update the relationships in the `.magento.app.yaml` file.

{:.procedure}
To downgrade a service version by renaming an existing service:

1. Rename the existing service in the `.magento/services.yaml` file and change the version.

   {:.bs-callout-warning}
   Renaming an existing service replaces it and deletes all data. If you need to retain the data, create a new service instead of renaming the existing one.

>    For example, to downgrade the MariaDB version for the _mysql_ service from version 10.3 to 10.2, change the existing service _name_ and _type_.

Reviewer: ```suggestion
   For example, to downgrade the MariaDB version for the _mysql_ service from version 10.3 to 10.2, change the existing  _service-id_ and _type_ configuration.
```

{:.procedure}
To downgrade a service version by renaming an existing service:

1. Rename the existing service in the `.magento/services.yaml` file and change the version.

   {:.bs-callout-warning}
   Renaming an existing service replaces it and deletes all data. If you need to retain the data, create a new service instead of renaming the existing one.

   For example, to downgrade the MariaDB version for the _mysql_ service from version 10.3 to 10.2, change the existing service _name_ and _type_.

   > Original `services.yaml` definition

     ```yaml
     mysql:
         type: mysql:10.3
         disk: 2048
     ```

   > New `services.yaml` definition

     ```yaml
     mysql2:
          type: mysql:10.2
          disk: 2048
     ```

1. Update the relationships in the `.magento.app.yaml` file.

   > Original `.magento.app.yaml` configuration

   ```yaml
   relationships:
       database: "mysql:mysql"
   ```

  > Updated `.magento.app.yaml` configuration

  ```yaml
  relationships:
      database: "mysql2:mysql"
  ```

1. Add, commit, and push your code changes.

{:.procedure}
To downgrade a service by creating an additional service:

> 1. Add the service definition to the `services.yaml` file for your project.

Reviewer: ```suggestion
1. Add an additional service definition to the `services.yaml` file for your project, _mysql2_ in the following example.
```
